include/oboe/Stream.h

```diff
@@ -177,16 +175,16 @@ class OboeStream : public OboeStreamBase {
      * @param timeoutNanoseconds Maximum number of nanoseconds to wait for completion.
      * @return The number of frames actually written or a negative error.
      */
-    virtual oboe_result_t write(const void *buffer,
+    virtual int32_t write(const void *buffer,
```

Comment:
IMO this method would be better off returning a `std::tuple` of `Result` and `int32_t`. This avoids the return type representing both the frames written (if positive) or an error (if negative).